cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceTaskContextProvider.java

```diff
@@ -229,13 +234,18 @@ public BasicMapReduceTaskContext load(ContextCacheKey key) throws Exception {
             taskType = MapReduceMetrics.TaskType.from(taskAttemptId.getTaskType());
             // if this is not for a mapper or a reducer, we don't need the metrics collection service
             metricsCollectionService = injector.getInstance(MetricsCollectionService.class);
+            options = new SimpleProgramOptions(options.getProgramId(), options.getArguments(),
+                                               new BasicArguments(
+                                                 RuntimeArguments.extractScope(
+                                                   "task", taskType.toString().toLowerCase(),
```

Comment:
I think task type can be `null`.

Reply:
this is inside the `if (MapReduceMetrics.TaskType.hasType(taskAttemptId.getTaskType())) {` condition and we perform `taskType = MapReduceMetrics.TaskType.from(taskAttemptId.getTaskType());` before creating this updated options. in MapreduceMetrics we would throw illegal argument exception if its not mapper or reducer.

Reply:
I see.